internal/controller/util/misc.go

```diff
@@ -254,3 +260,29 @@ func TrimToK8sResourceNameLength(name string) string {
 
 	return name
 }
+
+func GetJobName(namePrefix string, ownerName string) string {
+	return getShortenedResourceName(namePrefix, ownerName, JobNameMaxLength)
+}
+
+func GetServiceName(namePrefix string, ownerName string) string {
+	return getShortenedResourceName(namePrefix, ownerName, ServiceNameMaxLength)
+}
+
+func getShortenedResourceName(namePrefix string, ownerName string, maxLength int) string {
+	name := namePrefix + ownerName
+
+	if len(name) > maxLength {
+		// Use crc32 to hash the owner CR name
+		return namePrefix + GetHashedName(ownerName)
+	}
+
+	// No need to shorten, use original name
+	return name
+}
+
+// This mechanism is implemented in volsync operator
+// https://github.com/backube/volsync/pull/1519
+func GetHashedName(name string) string {
```

Comment:
To make this more clear, we can rename the function to `volsyncShortName()`. The Get prefix is not very helpful in most cases.

Reply:
Done

Reply:
The hash is specific to volsync and we cannot change it. If there was not dependency on volsync we could use a better hash algorithm.

Reply:
The function name should remain as it was originally. It is a utility function, even if it uses a less robust hashing algorithm. Besides, we're not encrypting passwords or anything sensitive. We are simply shortening the name, so using a simple algorithm is desirable.